packager/mpd/base/period.cc

```diff
@@ -216,16 +216,30 @@ bool Period::SetNewAdaptationSetAttributes(
     }
 
     if (media_info.video_info().has_playback_rate()) {
-      const AdaptationSet* trick_play_reference_adaptation_set =
-          FindOriginalAdaptationSetForTrickPlay(
-              media_info, content_protection_in_adaptation_set);
-      if (!trick_play_reference_adaptation_set) {
-        LOG(ERROR) << "Failed to find original AdaptationSet for trick play.";
-        return false;
+      std::string trick_play_reference_adaptation_set_key;
+      AdaptationSet* trick_play_reference_adaptation_set =
+          FindMatchingAdaptationSetForTrickPlay(
+              media_info, content_protection_in_adaptation_set,
+              &trick_play_reference_adaptation_set_key);
```

Comment:
The code LGTM with one optional nit: you may move this code outside of the "if else" block as it is used by both cases. 

Btw, have you verified that it addresses #732?

I'll take another look tomorrow and merge the code. Thanks very much for your help in addressing this issue.

Reply:
@kqyang Thanks for reviewing the code. I was able to test the 1 fps stream with trick_play_factor=1 as per issue #732 and the output mpd was correct.
I thought about removing the key declaration and call to FindMatchingAdaptationSetForTrickplay before the "if else" block but kept it for readability. Please let me know what you think.

Reply:
Sure, I feel moving it outside can be a bit more readable (the variables need to be renamed), but I am fine either ways.

Thanks for the hard work on this feature. Merging it now.